src/services/projects.js

```diff
@@ -604,19 +659,142 @@ module.exports = class ProjectsHelper {
+			if (model === common.TASKS && entityData.allow_evidences == common.TRUE) {
+				// Check if file types are selected
+				if (!entityData?.evidence_details?.file_types.length) {
+					return {
+						hasError: true,
+						error: utils.errorObject(common.BODY, common.FILE_TYPE, 'File type not selected'),
+					}
+				}
+
+				if (entityType.value === common.FILE_TYPE || entityType.value === common.MIN_NO_OF_EVIDENCES) {
+					fieldData = entityData.evidence_details[entityType.value]
+				}
+			}
+
+			// Check if the field is required
+			if (entityType.validations.required) {
+				let required = utils.checkRequired(entityType, fieldData)
+				if (!required) {
+					return {
+						hasError: true,
+						error: utils.errorObject(
+							sourceType,
+							entityType.value,
+							`${model} ${entityType.value} is required`
+						),
+					}
+				}
+			}
+
+			// Check if the entity has sub-entities
+			if (entityType.has_entities) {
+				let checkEntities = utils.checkEntities(entityType, fieldData)
+				if (!checkEntities.status) {
+					return {
+						hasError: true,
+						error: utils.errorObject(sourceType, entityType.value, checkEntities.message),
+					}
+				}
+			}
+
+			// Check regex pattern will check max length and special characters
+			if (entityType.validations.regex && fieldData) {
+				//validate learning resource validation
+				if (entityType.value === common.LEARNING_RESOURCE) {
+					for (let eachResource of fieldData) {
+						//validate the name and url is there
+						if (!eachResource.name || !eachResource.url) {
+							return {
+								hasError: true,
+								error: utils.errorObject(
+									sourceType,
+									common.LEARNING_RESOURCE,
+									`Required learning resource name and url in ${model}`
+								),
+							}
+						}
+						//validate the name
+						let validateName = utils.checkRegexPattern(entityMapping[common.NAME], eachResource.name)
+						if (validateName) {
+							return {
+								hasError: true,
+								error: utils.errorObject(
+									sourceType,
+									common.LEARNING_RESOURCE,
+									`Invalid learning resource name in ${model}`
+								),
+							}
+						}
+						//validate the url
+						let validateURL = utils.checkRegexPattern(
+							entityMapping[common.LEARNING_RESOURCE],
+							eachResource.url
+						)
+						if (validateURL) {
+							return {
+								hasError: true,
+								error: utils.errorObject(
+									sourceType,
+									common.LEARNING_RESOURCE,
+									`Invalid learning resource URL in ${model}`
+								),
+							}
+						}
+					}
+				} else {
+					let checkRegex = utils.checkRegexPattern(entityType, fieldData)
+					if (checkRegex) {
+						return {
+							hasError: true,
+							error: utils.errorObject(
+								sourceType,
+								entityType.value,
+								`${model} ${entityType.value} is invalid, please ensure it contains no special characters and does not exceed the character limit`
```

Comment:
@priyanka-TL Let's put a validationMessage column in entity type table and use that, I am very sure people would want to change this later. Field name can be added via here.